### PR TITLE
bugfix: container inspect/list should return image field right

### PR DESCRIPTION
**Pull request by fuweid**
### Ⅰ. Describe what this PR did

In container inspect, we replace the root.Image field by
root.Config.Image. We should keep it unchanged. And container list API
also returns root.ImageID field.

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

already add and update.

### Ⅳ. Describe how to verify it

CI

### CODE REVIEW

**Comment by Ace-Tang** [comment]
@fuweid , can you give some example to make it more readable.

**Comment by fuweid** [comment]
> can you give some example to make it more readable.

```
# vagrant @ ubuntu-xenial in ~/go/src/github.com/alibaba/pouch/test on git:bugfix_should_return_image_field_right o [11:41:20]
$ sudo pouch run -d busybox sleep 1000
3a30f16beac8edd5f934c269ffce526b17b6d254cff92f35d76688d2a9a14bfc

# vagrant @ ubuntu-xenial in ~/go/src/github.com/alibaba/pouch/test on git:bugfix_should_return_image_field_right o [11:41:35]
$ sudo pouch inspect 3a30f16beac8edd5f934c269ffce526b17b6d254cff92f35d76688d2a9a14bfc
[
    {
        "Args": null,
        "Created": "2018-09-12T03:41:34.565110419Z",
        "ExecIDs": null,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.overlayfs/snapshots/10/fs",
                "MergedDir": "/var/lib/pouch/containerd/state/io.containerd.runtime.v1.linux/default/3a30f16beac8edd5f934c269ffce526b17b6d254cff92f35d76688d2a9a14bfc/rootfs",
                "UpperDir": "/var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.overlayfs/snapshots/47/fs",
                "WorkDir": "/var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.overlayfs/snapshots/47/work"
            },
            "Name": "overlayfs"
        },
        "HostConfig": {
            "LogConfig": {
                "Type": "json-file"
            },
            "NetworkMode": "bridge",
            "OomScoreAdj": -500,
            "RestartPolicy": {
                "Name": "no"
            },
            "Runtime": "runc",
            "ShmSize": 0,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceWriteIOps": null,
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "CgroupParent": "default",
            "CpuCount": 0,
            "CpuPercent": 0,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpuShares": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "DeviceCgroupRules": null,
            "Devices": [],
            "IOMaximumBandwidth": 0,
            "IOMaximumIOps": 0,
            "IntelRdtL3Cbm": "",
            "KernelMemory": 0,
            "Memory": 0,
            "MemoryExtra": 0,
            "MemoryForceEmptyCtl": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": 0,
            "MemoryWmarkRatio": 0,
            "NanoCpus": 0,
            "OomKillDisable": false,
            "PidsLimit": 0,
            "ScheLatSwitch": 0,
            "Ulimits": null
        },
        "Id": "3a30f16beac8edd5f934c269ffce526b17b6d254cff92f35d76688d2a9a14bfc",
        "Image": "sha256:e1ddd7948a1c31709a23cc5b7dfe96e55fc364f90e1cebcde0773a1b5a30dcda",
        "Mounts": [],
        "Name": "3a30f1",
        "NetworkSettings": {
            "Networks": {
                "bridge": {
                    "Aliases": null,
                    "EndpointID": "78f97b4c27203ee316a22539dee6b1ee1b7377c7694e78b5525b28f3f0b01d8c",
                    "Gateway": "192.168.5.1",
                    "IPAddress": "192.168.5.2",
                    "IPPrefixLen": 24,
                    "Links": null,
                    "MacAddress": "02:42:c0:a8:05:02",
                    "NetworkID": "ef6b88a66ef4102e0472794ccc1ce42ee05741f3ab5eba71e483d66e88ed9376"
                }
            },
            "SandboxID": "c5d8fccc67f61050eb62d63322004388e2b8624c70879b84efc1f90cc37ba941",
            "SandboxKey": "/var/run/pouch/netns/c5d8fccc67f6",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null
        },
        "Snapshotter": {
            "Data": {
                "LowerDir": "/var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.overlayfs/snapshots/10/fs",
                "MergedDir": "/var/lib/pouch/containerd/state/io.containerd.runtime.v1.linux/default/3a30f16beac8edd5f934c269ffce526b17b6d254cff92f35d76688d2a9a14bfc/rootfs",
                "UpperDir": "/var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.overlayfs/snapshots/47/fs",
                "WorkDir": "/var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.overlayfs/snapshots/47/work"
            },
            "Name": "overlayfs"
        },
        "State": {
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Pid": 4613,
            "Running": true,
            "StartedAt": "2018-09-12T03:41:35.116527596Z",
            "Status": "running"
        },
        "Config": {
            "Cmd": [
                "sleep",
                "1000"
            ],
            "Entrypoint": null,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Hostname": "3a30f16beac8",
            "Image": "registry.hub.docker.com/library/busybox:latest",
            "OnBuild": null,
            "Shell": null
        }
    }
]
```

The `config.Image` means the image which user uses to create container, and `Image`always means the id of image. We should not change it.

**Comment by Ace-Tang** [comment]
@fuweid , sure, moby show like that.

File: apis/server/container_bridge.go [hunk 1]
```diff
@@ -134,6 +134,7 @@ func (s *Server) getContainers(ctx context.Context, rw http.ResponseWriter, req
 			ID:              c.ID,
 			Names:           []string{c.Name},
 			Image:           c.Config.Image,
+			ImageID:         c.Image,
```
Comment (HusterWan) [inline apis/server/container_bridge.go]: should we also change the logic in `removeImage` when compare the image

**Comment by HusterWan** [comment]
LGTM

File: apis/server/container_bridge.go (hunk 1 above)
Comment (fuweid) [reply apis/server/container_bridge.go]: I check the logic and found that we should use imageID to do the compare when we remove the image. There are some cases during removing image:

* rmi shortID
* rmi imageName which share same id with other one

Therefore, we use imageID to do compare to check container is using the image or not.

**Pull request merged**